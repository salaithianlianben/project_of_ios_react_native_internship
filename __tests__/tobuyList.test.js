import React from 'react';
import renderer from 'react-test-renderer';
import ToBuyList from '../src/screens/tobuyList';

test('render of To Buy List is correclty', () => {
  const rendered = renderer.create(<ToBuyList />).toJSON();
  expect(rendered).toMatchSnapshot();
  expect(rendered).toBeTruthy();
});

it('function and state on change search text box and add new item',()=>{
    let tobuyListData = renderer.create(<ToBuyList/>).getInstance();

    tobuyListData.onHandleChangeSearchTextBox("H")

    expect(tobuyListData.state.search_value).toEqual("H")

    // after adding new item into tobuyList, checking the length of tobuyList array of object

    tobuyListData.addToBuyList()

    expect(tobuyListData.state.to_buy_data.length).toEqual(4)

    expect(tobuyListData.state.to_buy_data.length).toBeGreaterThan(tobuyListData.state.to_buy_data.length - 1);

    // checking type of generated item name
    expect(typeof tobuyListData.generateRandomText()).toBe("string")
});



const toBuyListData = [
    {
        id: 1,
        item_name: "Hair Spray",
    },
    {
        id: 2,
        item_name: "Lotion",
    },
    {
        id: 3,
        item_name: "Nature Pack"
    }
];

test('It should contain ids 1 and 2', () => {
  expect(toBuyListData).toEqual(
    expect.arrayContaining([
      expect.objectContaining({id: 1}),
      expect.objectContaining({id: 2})
    ])
  );
});

test('function filter value from data',()=>{
    let tobuyListData = renderer.create(<ToBuyList/>).getInstance();

    expect( tobuyListData.getFilterData(toBuyListData,"H") ).toEqual( 
        expect.arrayContaining([ 
          expect.objectContaining(
            {
                id: 1,
                item_name: "Hair Spray",
            }
          )
        ])
      );
});