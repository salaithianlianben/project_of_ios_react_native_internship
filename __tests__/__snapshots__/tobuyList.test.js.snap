// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render of To Buy List is correclty 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#A5A5A5",
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Search"
      style={
        Object {
          "borderColor": "#A5A5A5",
          "borderWidth": 1,
          "flex": 1,
          "paddingHorizontal": 10,
        }
      }
      value=""
    />
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 40,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": 1,
          "item_name": "Hair Spray",
        },
        Object {
          "id": 2,
          "item_name": "Lotion",
        },
        Object {
          "id": 3,
          "item_name": "Nature Pack",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "borderBottomColor": "#A5A5A5",
              "borderBottomWidth": 1,
              "padding": 10,
            }
          }
        >
          <Text>
            Hair Spray
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "borderBottomColor": "#A5A5A5",
              "borderBottomWidth": 1,
              "padding": 10,
            }
          }
        >
          <Text>
            Lotion
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "borderBottomColor": "#A5A5A5",
              "borderBottomWidth": 1,
              "padding": 10,
            }
          }
        >
          <Text>
            Nature Pack
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#A5A5A5",
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Search"
      style={
        Object {
          "borderColor": "#A5A5A5",
          "borderWidth": 1,
          "flex": 1,
          "paddingHorizontal": 10,
        }
      }
      value=""
    />
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 40,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": 1,
          "item_name": "Hair Spray",
        },
        Object {
          "id": 2,
          "item_name": "Lotion",
        },
        Object {
          "id": 3,
          "item_name": "Nature Pack",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "borderBottomColor": "#A5A5A5",
              "borderBottomWidth": 1,
              "padding": 10,
            }
          }
        >
          <Text>
            Hair Spray
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "borderBottomColor": "#A5A5A5",
              "borderBottomWidth": 1,
              "padding": 10,
            }
          }
        >
          <Text>
            Lotion
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "borderBottomColor": "#A5A5A5",
              "borderBottomWidth": 1,
              "padding": 10,
            }
          }
        >
          <Text>
            Nature Pack
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
